### PR TITLE
node|cmd: Make `StateService` required on node for all node types

### PR DESCRIPTION
**Pull request by renaynay**
Make `StateService` required on `node`. As we only have the `CoreAccess` implementation of `StateAccessor` for now, all nodes will be required to pass a `--core.remote` endpoint on init/start or else construction will fail. Eventually when we implement `P2PAccess`, a core endpoint will be optional, but `StateService` will still be required.

### CODE REVIEW

**Review by Wondertan — CHANGES_REQUESTED** [review]
Let's extract the .girignore change into a separate PR which also includes change in makefile which make it build not in the root but into ‘build’ folder

**Comment by adlerjohn** [comment]
Do you mean temporary build files, or the final binary? The final binary should probably be in `bin` rather than `build`.

**Comment by Wondertan** [comment]
> Do you mean temporary build files, or the final binary? The final binary should probably be in `bin` rather than `build`.

@adlerjohn I assume that ‘make build’ is for temporary builds. We don't store final builds in the repo.

**Review by liamsi — APPROVED** [review]

File: cmd/celestia/cmd.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+
+	cmdnode "github.com/celestiaorg/celestia-node/cmd"
+)
+
+func parseFlags(cmd *cobra.Command, args []string) error {
```
[the remark this answers is not in the document]
Comment (renaynay) [reply cmd/celestia/cmd.go:9]: I mean the same way we have `baseComponents`, we can have `baseFlags` func parsing for base flags and then have the remainder custom to node type flags additionally?

File: cmd/celestia/full.go [hunk 2]
```diff
@@ -48,7 +55,12 @@ var fullCmd = &cobra.Command{
 			return err
 		}
 
-		err = cmdnode.ParseHeadersFlags(cmd, env)
+		err = cmdnode.ParseCoreFlags(cmd, env)
+		if err != nil {
+			return err
+		}
+
+		err = cmdnode.ParseTrustedHashFlags(cmd, env)
```
Comment (Wondertan) [inline cmd/celestia/full.go:63]: Full node also needs trusted peer thus we should allow it to be modified

File: cmd/celestia/light.go [hunk 3]
```diff
@@ -54,17 +54,17 @@ var lightCmd = &cobra.Command{
...
 		}
 
-		err = cmdnode.ParseMiscFlags(cmd)
+		err = cmdnode.ParseTrustedHashFlags(cmd, env)
 		if err != nil {
 			return err
 		}
 
-		err = cmdnode.ParseCoreFlags(cmd, env)
+		err = cmdnode.ParseMiscFlags(cmd)
```
Comment (Wondertan) [inline cmd/celestia/light.go:67]: * Why order change?
* Same here about trusted peer. Why did you remove it?

**Review by Wondertan — CHANGES_REQUESTED** [review]

**Review by Wondertan — APPROVED** [review]

**Pull request merged**